UniWatch/Controllers/LectureController.cs

```diff
@@ -0,0 +1,297 @@
+
+            if (lecture == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest, $"No lecture with id {lvm.LectureId}");
+            }
+
+            var updates = new List<UpdateLectureItem>(lecture.Attendance.Count);
+
+            updates.AddRange(from attendance in lecture.Attendance
+                             let present = lvm.LectureItems
+                                .FirstOrDefault(item => attendance.Student.Id == item.StudentId)?.Present ?? null
+                             where present.HasValue && attendance.Present != present.Value
+                             select new UpdateLectureItem
+                             {
+                                StudentId = attendance.Student.Id,
+                                LectureId = lecture.Id,
+                                Present = present.Value
+                             }
+            );
+
+            _manager.LectureManager.Update(lecture.Id, updates);
+
+            return RedirectToAction("Override", "Lecture", new { lectureId = lecture.Id });
+        }
+
+        #endregion
+
+        #region Create
+
+        [HttpGet]
+        public ActionResult Create(int classId)
+        {
+            var @class = _manager.ClassManager.GetById(classId);
+
+            if(@class == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest, $"No class with id {classId}");
+            }
+
+            var lecture = new Lecture() { Class = @class };
+
+            return View(lecture);
+        }
+
+        [HttpPost]
+        public ActionResult Create(Lecture lecture)
+        {
+            if(lecture.Class == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest, "No class id");
+            }
+
+            if(Request.Files.Count == 0)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest, "No files selected");
+            }
+
+            var fileStreams = new List<Stream>(Request.Files.Count);
+
+            foreach(HttpPostedFileBase file in Request.Files)
+            {
+                if(file.ContentLength == 0 /*&& file.ContentLength >= MAX_SIZE*/)
+                {
+                    return new HttpStatusCodeResult(HttpStatusCode.BadRequest, $"Selected file {file.FileName} is out of bounds");
+                }
+
+                fileStreams.Add(file.InputStream);
+            }
+
+            _manager.LectureManager.RecordLecture(lecture.Class.Id, fileStreams);
+            // TODO: while the system is creating the lecture and taking attendance, display a load screen
+
+            return RedirectToAction("Index", "Lecture", new { classId = lecture.Class.Id });
+        }
+
+        #endregion
+
+        #region Delete
+
+        [HttpGet]
+        public ActionResult Delete(int lectureId)
+        {
+            var lvm = GetLectureViewModel(lectureId);
```

Comment:
This is doing an unnecessary amount of work. To delete a class, you generate the update view model and then not use it? Delete just needs the lecture